guillotina/contrib/cache/utility.py

```diff
@@ -0,0 +1,223 @@
+                    val = serialize.loads(val)
+                    self._memory_cache[key] = val
+        except Exception:
+            logger.warning('Error getting cache value', exc_info=True)
+
+    def get_size(self, value):
+        if isinstance(value, dict):
+            if 'state' in value:
+                return len(value['state'])
+        if isinstance(value, list) and len(value) > 0:
+            # if its a list, guesss from first gey the length, and
+            # estimate it from the total lenghts on the list..
+            return getsizeof(value[0]) * len(value)
+        if type(value) in _basic_types:
+            return getsizeof(value)
+        return _default_size
+
+    # Set a object from cache
+    async def set(self, key, value):
+        try:
+            size = self.get_size(value)
+            self._memory_cache.set(key, value, size)
+            if self._obj_driver is not None:
+                await self._obj_driver.set(
+                    CACHE_PREFIX + key,
+                    serialize.dumps(value),
+                    expire=self._settings.get('ttl', 3600))
+            logger.info('set {} in cache'.format(key))
+        except Exception:
+            logger.warning('Error setting cache value', exc_info=True)
+
+    @profilable
+    # Delete a set of objects from cache
+    async def delete_all(self, keys):
+        delete_keys = []
+        for key in keys:
+            delete_keys.append(CACHE_PREFIX + key)
+            if key in self._memory_cache:
+                del self._memory_cache[key]
+        if len(delete_keys) > 0 and self._obj_driver is not None:
+            await self._obj_driver.delete_all(delete_keys)
+
+    # Delete a set of objects from cache
+    async def delete(self, key):
+        try:
+            if key in self._memory_cache:
+                del self._memory_cache[key]
+            await self._obj_driver.delete(key)
+        except Exception:
+            logger.warning('Error removing from cache', exc_info=True)
+
+    # Clean all cache
+    async def clear(self):
+        try:
+            self._memory_cache.clear()
+            if self._obj_driver is not None:
+                await self._obj_driver.flushall()
+            logger.info('Cleared cache')
+        except Exception:
+            logger.warning('Error clearing cache', exc_info=True)
+
+    @profilable
+    # Called by the subscription to invalidations
+    async def invalidate(self, *, data=None, sender=None):
+        try:
+            msg = serialize.loads(data)
+        except (TypeError, pickle.UnpicklingError):
+            logger.warning("Invalid invalidation message")
+            pass
+
+        assert isinstance(msg, dict)
+        assert 'tid' in msg
+        assert 'keys' in msg
+        if msg['tid'] in self._ignored_tids:
+            # on the same thread, ignore this sucker...
+            self._ignored_tids.remove(msg['tid'])
+            return
+
+        mem_cache_obj = memcache.get_memory_cache()
```

Comment:
should we use `self._memory_cache` instead?

Reply:
yups